src/main/java/kz/zhelezyaka/dao/PetDAO.java

```diff
@@ -0,0 +1,151 @@
+package kz.zhelezyaka.dao;
+
+import kz.zhelezyaka.connection.ConnectionPool;
+import kz.zhelezyaka.entity.Pet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.NamingException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PetDAO {
```

Comment:
```suggestion
public class PetRepositoryImpl {
```
Создаем базовый интерфейс что-то типа этого:
```java
public interface BaseRepository<T extends BaseEntity> {

    T create(T entity);

    List<T> read();

    T update(T entity);

    void delete(Long id);
}
```
От него наследуются интерфейсы для каждого класса репозитория. Соответствующие классы репозитории реализуют свои интерфейсы. Конкретные интерфейсы репозитории содержат абстрактные методы которых нету в BaseRepository